First-Peoples-Cultural-Council/fv-be
# Fw 4378 all site search api

### Description of Changes
Addition of a basic view to implement site-wide search with just two parameters as of now.
This PR contains basic setup for elastic search addition to the backend such as indexing, documents and one view with one parameter.
[realted wiki page](https://firstvoices.atlassian.net/wiki/spaces/FIR/pages/275251209/ElasticSearch+Draft

### Checklist
- [x] README / documentation has been updated if needed
- [x] PR title / commit messages contain Jira ticket number if applicable
- ~Tests have been updated / created if applicable~
- ~Admin Panel has been updated if models have been added or modified~
- ~Migrations have been updated and committed if applicable~

### Reviewers Should Do The Following:
- [x] Review the code changes here
- [x] Pull the branch and test locally (Refer to instructions below for local testing) 

### Additional Notes

To test locally:

1. Ensure that elastic-search server is running locally. Instructions can be found in the readme to get the docker container running.
2. Add any new words and they will be indexed automatically. Updating or deleting should also be tested.
3. The following url can be used with the q parameter
url: http://localhost:8000/api/1.0/search/?q=wordTitle
```
q: enter a search query which can be matched with the title field in the dictionaryEntry model as of now.
```
4. Verify that the search results returned contains the query param.
